File: leetcode/groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        for i in strs:
            li = list(i)
            li.sort()
            tmp = ''.join(li)
            
            if tmp in dic: dic[tmp].append(i)
            else: dic[tmp] = [i]
        
        
        ans = []
        for i in dic.values(): ans.append(i)
            
        return ans

File: leetcode/groupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs){
        
        unordered_map<string, vector<string>> myMap;
        
        for(string& str : strs) {
            string tmp = str;
            sort(tmp.begin(), tmp.end());
            myMap[tmp].push_back(str);
        }
        
        vector<vector<string>> ans;
        for(auto& it : myMap) ans.push_back(it.second);
        
        return ans;
    }
};